### PR TITLE
Draft: SC4-1-2-button-has-name

### PR DESCRIPTION
**Pull request by jkodu**
Draft rule as discussed in issue - 
https://github.com/auto-wcag/auto-wcag/issues/100

### CODE REVIEW

File: _drafts/SC4-1-2-button-has-name.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+---
+name: Buttons have an accessible name
+description: |
+  Each button element has an accessible name
+
+success_criterion:
+- 4.1.2 # Name, Role, Value
+
+test_aspects: # Remove what is not applicable
```
Comment (Brynanders) [inline _drafts/SC4-1-2-button-has-name.md]: Remove -> # Remove what is not applicable

File: _drafts/SC4-1-2-button-has-name.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+The rule applies to any element with the role of `button`, except for `input` elements of `type="image"` that are [visible on the page][] or [exposed to assistive technologies][].
+
+### Expectation
+
+Each target element has an [accessible name][] that is [non-empty][].
+
+## Assumptions
+
+- The rule assumes that all buttons are user interface components as defined by WCAG 2. When the button role is used incorrectly, the assumption may not be true.
```
Comment (Brynanders) [inline _drafts/SC4-1-2-button-has-name.md]: Do we need the second sentence? The assumption is already defined in the first sentence.

File: _drafts/SC4-1-2-button-has-name.md (hunk 1 above)
Comment (jkodu) [reply _drafts/SC4-1-2-button-has-name.md]: This is a comment (anything after hash #), and will not be shown or generated in the end result.
Either- way removed it.

File: _drafts/SC4-1-2-button-has-name.md (hunk 2 above)
Comment (jkodu) [reply _drafts/SC4-1-2-button-has-name.md]: You are right. First sentence does the trick

**Comment by jkodu** [comment]
@ShadowBB 
Could I lean on your expertise for an aria-label test case? Just drop a note here and I shall amend the draft.

**Comment by WilcoFiers** [comment]
@Brynanders @ShadowBB please **APPROVE** if you have no problem with publishing the draft.

**Comment by ShadowBB** [comment]
@jkodu 

Among the passed test cases:

```html
<button aria-label="My button"></button>
```

Should be enough.

**Comment by cpandhi** [comment]
I approve.

File: _drafts/SC4-1-2-button-has-name.md (hunk 2 above)
Comment (cpandhi) [reply _drafts/SC4-1-2-button-has-name.md]: Is it necessary to clarify in th Applicability section that any element with the role of `button` also include HTML elements with implied role of button? example 
``input `type=submit`` has role=button.

**Pull request merged**